Converters/Autocad/Speckle.Converters.AutocadShared/Extensions/ListExtensions.cs

```diff
@@ -25,17 +25,91 @@ public static class ListExtensions
     return points3d;
   }
 
+  /// <summary>
+  /// Converts polyline vertex data to Point3d in OCS, supporting both 2D and 3D formats.
+  /// 2D format: XY pairs already in OCS (from Point2d)
+  /// 3D format: XYZ triplets in external coordinates that need OCS transformation
+  /// </summary>
+  public static List<AG.Point3d> ConvertPolylineValueToPoint3dInOcs(
+    this List<double> pointList,
+    AG.Vector3d normal,
+    double elevation,
+    double conversionFactor = 1,
+    List<double>? bulges = null,
+    List<double?>? tangents = null
+  )
+  {
+    bool is2DPointList = IsPointList2D(pointList, bulges, tangents);
+
+    if (is2DPointList)
+    {
+      // 2D format: XY pairs are already in OCS, just need to add elevation as Z
```

Comment:
Unfortunately, can't use the elevation for converting 2d points to 3d, since the elevation is actually the elevation of the plane of the polyline on the world z axis. Different vertices can have different z values depending on their position in their OCS plane